src/crypto/CHIPCryptoPAL.h

```diff
@@ -36,6 +36,8 @@
 namespace chip {
 namespace Crypto {
 
+const size_t kMax_x509_Certificate_Length = 1024;
```

Comment:
Where does this come from?

Reply:
So it appears the CSR (or elements thereof) is limited to 1024 bytes.  I'm not seeing the actual certificate size limit in the spec though.  I assume there has to be a limit.  I'm just looking for confirmation that this is correctly set to that limit.

Reply:
So now I see that this was changed to 600 bytes.  And indeed, the spec does clarify that the root CA certificate must be 600 or fewer bytes.  But do we have different limits for ICA or NOC?  This part wasn't clear to me.

Reply:
As far as I know, only root certificates have a defined max length, but this ensures standard compact sizing when dealing with a full certificate chain.

Reply:
So I guess then I don't understand how we're supposed to succeed in this PR.  It looks like the PR attempts to also store CA and leaf certs within kMax_x509_Certificate_Length bytes, even though this is only guaranteed to fit the root cert.

This doesn't seem like it will work. Am I missing something?

Reply:
The 600 limit is for DAC and PAI and PAA which are DER.

For opcerts, RCAC, ICAC and NOC are supposed to be TLV certs and max 400 bytes.